node/pkg/watchers/near/metrics.go

```diff
@@ -0,0 +1,125 @@
+package near
+
+import (
+	"context"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/p2p"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/certusone/wormhole/node/pkg/readiness"
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+)
+
+type eventType int
+
+const (
+	EVENT_FINALIZED_CACHE_MISS eventType = iota
+	EVENT_NEAR_MESSAGE_CONFIRMED
+	EVENT_NEAR_API_HTTP_ERR // NEAR API returned a status code other than 200
+	EVENT_NEAR_WATCHER_TOO_FAR_BEHIND
+)
+
+func (e *Watcher) runMetrics(ctx context.Context) error {
+	logger := supervisor.Logger(ctx)
+
+	reg := prometheus.NewRegistry()
+
+	wormholeMsgCounter := 0
+	var wormholeMsgTotalTime time.Duration = 0
+
+	currentNearHeight := promauto.With(reg).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "wormhole_near_current_height",
+			Help: "Height of the highest block that has been processed. (Transactions from prior blocks may still be waiting).",
+		})
+
+	wormholeTxAvgDuration := promauto.With(reg).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "wormhole_near_tx_avg_duration",
+			Help: "Average duration it takes for a wormhole message to be processed in milliseconds",
+		})
+
+	txQuequeLen := promauto.With(reg).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "wormhole_near_tx_queque",
+			Help: "Current Near transaction processing queque length",
+		})
+
+	chunkQuequeLen := promauto.With(reg).NewGauge(
+		prometheus.GaugeOpts{
+			Name: "wormhole_near_chunk_queque",
+			Help: "Current Near chunk processing queque length",
+		})
+
+	nearMessagesConfirmed := promauto.With(reg).NewCounter(
+		prometheus.CounterOpts{
+			Name: "wormhole_near_observations_confirmed_total",
+			Help: "Total number of verified Near observations found",
+		})
+
+	nearFinalizedCacheMisses := promauto.With(reg).NewCounter(
+		prometheus.CounterOpts{
+			Name: "wormhole_near_finalized_cache_misses",
+			Help: "Total number of verified Near observations found",
```

Comment:
Maybe add "in cache" on the end to distinguish from line 62?

Reply:
thanks